src/Components/BehaviorsModal/Internal/Widgets/DataHistoryWidget/DataHistoryWidget.tsx

```diff
@@ -1,47 +1,178 @@
+import { useTimeSeriesData } from '../../../../../Models/Hooks/useTimeSeriesData';
+import {
+    IDataHistoryTimeSeries,
+    IDataHistoryWidget,
+    IDataHistoryWidgetConfiguration
+} from '../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import HighChartsWrapper from '../../../../HighChartsWrapper/HighChartsWrapper';
+import { IHighChartSeriesData } from '../../../../HighChartsWrapper/HighChartsWrapper.types';
 import { BehaviorsModalContext } from '../../../BehaviorsModal';
+import { dataHistoryClassNames } from './DataHistoryWidget.styles';
 
 interface IProp {
     widget: IDataHistoryWidget;
 }
 
-export const DataHistoryWidget: React.FC<IProp> = ({ widget }) => {
-    const { mode } = useContext(BehaviorsModalContext);
+const DataHistoryWidget: React.FC<IProp> = ({ widget }) => {
     const {
         displayName,
         connectionString,
         timeSeries,
         chartOptions
     } = widget.widgetConfiguration;
 
+    const { adapter, twins, mode } = useContext(BehaviorsModalContext);
+    const twinIdPropertyMap = getTwinIdPropertyMap(timeSeries, twins);
+    const isRequestSent = useRef(false);
+
+    const {
+        query,
+        deeplink,
+        data,
+        fetchTimeSeriesData,
+        isLoading
+    } = useTimeSeriesData({
+        adapter,
+        connectionString,
+        quickTimeSpan: chartOptions.defaultQuickTimeSpan,
+        twins: twinIdPropertyMap
+    });
+
+    useEffect(() => {
+        if (
+            mode === BehaviorModalMode.viewer &&
+            query &&
+            (adapter || connectionString) &&
+            !isRequestSent.current &&
+            twinIdPropertyMap
+        ) {
+            fetchTimeSeriesData();
+            isRequestSent.current = true;
+        }
+    }, [
+        adapter,
+        query,
+        connectionString,
+        twinIdPropertyMap,
+        chartOptions,
+        mode
+    ]);
+
+    const placeholderTimeSeriesData: Array<
+        Array<TimeSeriesData>
+    > = useMemo(() => {
+        // placeholder timeseries data to be used in preview mode, need to memoize not to change the chart plot area unless time span or series length changes
+        const toInMillis = Date.now();
+        const fromInMillis = toInMillis - chartOptions.defaultQuickTimeSpan;
+        return timeSeries.map(() =>
+            Array.from({ length: 5 }, () => ({
+                timestamp: Math.floor(
+                    Math.random() * (toInMillis - fromInMillis + 1) +
+                        fromInMillis
+                ),
+                value: Math.floor(Math.random() * 500)
+            })).sort(
+                (a, b) => (a.timestamp as number) - (b.timestamp as number)
+            )
+        );
+    }, [chartOptions.defaultQuickTimeSpan, timeSeries.length]);
+
+    const highChartSeriesData: Array<IHighChartSeriesData> = useMemo(
+        () =>
+            mode === BehaviorModalMode.preview
+                ? generatePlaceholderHighChartsData(
```

Comment:
preview looks like the same as the real thing except the data. mock data and real data are in the same format and we feed the highchart wrapper component in the same way (in  a format of IHighChartSeriesData). I just need to generate mock data here.

Reply:
not sure if I quite understand what you are asking. There is no data fetching at this point, so there is no reason to update the mock data or something since there is no data pipeline happening. What I do here is same as what I do in ValueWidget with placeholder values.